ray-operator/main.go

```diff
@@ -246,13 +248,21 @@ func main() {
 			rayServiceMetricsManager,
 		)
 	}
+
+	if config.EnableBatchScheduler {
```

Comment:
```suggestion
	if config.EnableBatchScheduler || config.BatchScheduler != "" {
```

The current logic checks both the deprecated EnableBatchScheduler flag and the new BatchScheduler field, which is not fully backward compatible.

Since `EnableBatchScheduler` is now deprecated, we should still keep it for backward compatibility, but only one of the two should be set.
We can simply check either `BatchScheduler.Name` or `batchScheduler.enable`.

Ref: https://github.com/ray-project/kuberay/pull/2300/files

Reply:
Good catch, thanks @win5923 !